src/common/snippets/src/lowered/pass/solve_buffer_memory.cpp

```diff
@@ -14,73 +15,144 @@ namespace snippets {
 namespace lowered {
 namespace pass {
 
-std::vector<ov::MemorySolver::Box> SolveBufferMemory::init_boxes(const AllocateBuffers::BufferClusters& buffer_clusters) {
-    std::vector<ov::MemorySolver::Box> boxes;
-    const auto count = static_cast<int>(buffer_clusters.size());
-    for (int i = 0; i < count; i++) {
-        ov::MemorySolver::Box box = { std::numeric_limits<int>::max(), 0, 0, i };
-        int64_t box_size = 0;
-        for (const auto& buffer_expr : buffer_clusters[i]) {
-            int e_start = 0, e_finish = 0;
-            const auto buffer = ov::as_type_ptr<ov::snippets::op::Buffer>(buffer_expr->get_node());
-            OPENVINO_ASSERT(buffer != nullptr, "BufferSolver expects Buffer ops in clusters");
-
-            // life finish time - order of LoopEnd / MemoryAccess ops
-            const auto& buffer_outs = buffer_expr->get_output_port_connectors();
-            for (const auto& buffer_out : buffer_outs) {
-                const auto consumers = buffer_out->get_consumers();
-                for (const auto& consumer : consumers) {
-                    const auto consumer_order = static_cast<int>(ov::snippets::pass::GetTopologicalOrder(consumer.get_expr()->get_node()));
-                    e_finish = std::max(e_finish, consumer_order);  // the last consumer
-                }
+std::pair<LinearIR::container, LinearIR::container> SolveBufferMemory::extract_static_and_dynamic_buffers(const LinearIR::container& buffer_expressions) {
```

Comment:
Since the first moment I saw `m_buffer_expressions` in LIR, I anticipated that we eventually have to split it to static and dynamic. And here we are 😁 
Then a logical question comes up: why don't we do it straight away and create `m_static_buffer_expressions` and `m_dynamic_buffer_expressions`? Because we have to split them again in runtime_configurator, don't we? 

Also, please keep in mind that all these m_buffer_* should probably be moved to a dedicated memory manager eventually.

Reply:
Discussed offline: we should support valid state of these containers during all pipeline. However, it's not possible at this moment. Should be rewritten in bounds of https://jira.devtools.intel.com/browse/CVS-143395